src/main/java/com/savvato/basemobileapp/controllers/UserAPIController.java

```diff
@@ -48,43 +51,31 @@ public ResponseEntity createUser(@RequestBody @Valid UserRequest req) {
 			return new ResponseEntity(iae.getMessage(), HttpStatus.BAD_REQUEST);
 		}
 	}
-
-
-	// TODO: figure a way to have these enddpoints not be publicly accessible, but also not require a user to be logged in.
-	// 			a secret, but general purpose for-the-app login.
-
-
-	// api/public/user/isUsernameAvailable?q=sample
-	@RequestMapping(value = { "/api/public/user/isUsernameAvailable" })
-	public boolean isUsernameAvailable(@RequestParam("q") String queryStr) {
-		return this.ur.findByName(queryStr).isPresent() == false;
-	}
-
-	// api/public/user/isPhoneNumberAvailable?q=7205870001
-	@RequestMapping(value = { "/api/public/user/isPhoneNumberAvailable" })
-	public boolean isPhoneNumberAvailable(@RequestParam("q") String queryStr) {
-		Optional<List<User>> opt = this.ur.findByPhone(queryStr);
-
-		if (opt.isPresent())
-			return opt.get().size() == 0;
-		else
-			return true;
-	}
-
-	// api/public/user/isEmailAddressAvailable?q=anAddress@domain.com
-	@RequestMapping(value = { "/api/public/user/isEmailAddressAvailable" })
-	public boolean isEmailAddressAvailable(@RequestParam("q") String queryStr) {
-		return this.ur.findByEmail(queryStr).isPresent() == false;
-	}
-
+	
 	// api/public/user/isUserInformationUnique?name=sample&phone=7205870001&email=anAddress@domain.com
 	@RequestMapping(value = { "/api/public/user/isUserInformationUnique" })
-	public String isUserInformationUnique(@RequestParam("name") String username, @RequestParam("phone") String phone, @RequestParam("email") String email) {
-		if (!isUsernameAvailable(username)) return "{\"response\": \"username\"}";
-		if (!isPhoneNumberAvailable(phone)) return "{\"response\": \"phone\"}";
-		if (!isEmailAddressAvailable(email)) return "{\"response\": \"email\"}";
+	public ResponseEntity<GenericResponseDTO> isUserInformationUnique(@RequestParam("name") String username, @RequestParam("phone") String phone, @RequestParam("email") String email) {
+		GenericResponseDTO genericResponseDTO = GenericResponseDTO.builder().build();
+		if (!isUserDataAvailable(UserData.USER_NAME,username)) {
+			genericResponseDTO.responseMessage = "username";
+		} else if (!isUserDataAvailable(UserData.USER_PHONE,phone)) {
+			genericResponseDTO.responseMessage = "phone number";
+		} else if (!isUserDataAvailable(UserData.USER_EMAIL,email)) {
+			genericResponseDTO.responseMessage = "email";
+		} else {
+			genericResponseDTO.responseMessage = "true";
+		}
+		return ResponseEntity.status(HttpStatus.OK).body(genericResponseDTO);
+	}
+
+	private boolean isUserDataAvailable(UserData typeOfData, String data){
+		boolean rtn = switch(typeOfData) {
```

Comment:
This is a very pretty refactor. Well done.

Reply:
Thanks! I'm trying to get more comfortable with the new switch statement, and this seemed like a good use case. (As long as I remember how to handle returned optionals haha)